kolibri/plugins/base.py

```diff
@@ -131,6 +142,31 @@ def api_url_module(self):
         """
         return self._return_module("api_urls")
 
+    def settings_module(self):
+        """
+        Return a settings module, containing Django settings that this
+        module wants to apply.
+
+        For settings that take a tuple or list, these will be appended to the value from
+        the base settings module set through conventional Django means.
+
+        By default settings.py will be automatically discovered, if you wish to
+        override this behaviour, override this method.
+        """
+        return self._return_module("settings")
+
+    def options_module(self):
+        """
+        Return an options module, containing a config spec as the 'option_spec' value.
+
+        These options should not override the core config spec, but may specify only a new
+        default value for a core config spec option.
+
+        By default options.py will be automatically discovered, if you wish to
+        override this behaviour, override this method.
+        """
+        return self._return_module("options")
+
```

Comment:
What if instead we had a separate options file for persistent plugin configuration? `plugin_options.ini` or similar?

Reply:
We could do - but it seemed reasonable to allow plugins to add to the core options.ini (but again, as with settings, not allow them to overwrite - the only exception being to let them change default values).